esteid/compat.py

```diff
@@ -0,0 +1,76 @@
+"""
+Compatibility utilities that provide signed container info
+in the same format as the DDS service did.
+"""
+import base64
+import hashlib
+from esteid.digidocservice.types import ResponderCertificate, SignedDocInfo
+from esteid.bdoc2.container import BDoc2File
+
+
+def signature_info(xml_signature):
+    subject_cert = xml_signature.get_certificate()
+    cert_asn1 = subject_cert.asn1
+    personal = cert_asn1.subject.native
+    validity = cert_asn1.native['tbs_certificate']['validity']
+
+    return {
+        'id': xml_signature._get_node('ds:Signature').attrib['Id'],
+        'signing_time': xml_signature._get_node('xades:SigningTime').text,
+        'status': True,
+        'signature_production_place': None,
+        'signer': {
+            'certificate': {
+                'issuer': xml_signature._get_node('xades:SigningCertificate//ds:X509IssuerName').text,
+                'issuer_serial': xml_signature._get_node('xades:SigningCertificate//ds:X509SerialNumber').text,
+                'policies': [],
+                'subject': personal['common_name'],
+                'valid_from': validity['not_before'].strftime('%Y-%m-%dT%H:%M:%SZ'),
+                'valid_to': validity['not_after'].strftime('%Y-%m-%dT%H:%M:%SZ'),
+            },
+            'full_name': '%s %s' % (personal['given_name'], personal['surname']),
+            'id_code': personal['serial_number'].split('-')[-1],
+        },
+        'confirmation': {
+            # if ever needed
+            'produced_at': xml_signature._get_node('xades:SigningTime').text,
+            'responder_id': 'OCSP',
+            'responder_certificate': ResponderCertificate(issuer='', valid_from='', valid_to='', issuer_serial='', subject=''),
```

Comment:
Is this not available? If not please add a comment here indicating why it is empty.

Reply:
Well honestly I just copied it over from the other project it was originally used in. I guess, it is present in the signature data. The question is, what is the intent, and do we need this piece at all, because this info structure is just trying to mimic the DDS behavior (which is about to become deprecated), but I'm not sure if anyone will ever make use of the OCSP responder certificate from this info structure. In the worst case, it's still available from the container/signature API. As of now, it's just a poorly structured (_flat is better than nested_) piece of duplicate information.